chall09/estina.c

```diff
@@ -0,0 +1,96 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   estina.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: estina <estina@student.42madrid.com>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/05/09 19:45:09 by estina            #+#    #+#             */
+/*   Updated: 2020/05/10 15:34:40 by estina           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <ctype.h>
+
+int		check_input(char *seconds)
+{
+	int		i;
+
+	if (!*seconds)
+		return (0);
+	i = strlen(seconds) - 1;
+	while (isspace(seconds[i]))
+		i--;
+	while (i >= 0)
+	{
+		if (!isdigit(seconds[i]))
+			return (0);
+		i--;
+	}
+	return (1);
+}
+
+unsigned long long	transform_time(void *y, int z)
+{
+	unsigned long long	x;
+
+	x = *((unsigned long long *)y) / z;
+	*((unsigned long long *)y) %= z;
+	return (x);
+}
+
+void	append_time(unsigned long long time, char *one, char *more, char *n, char **str)
+{
+	char	buf[13];
+
+	if (time)
+	{
+		sprintf(buf, "%llu", time);
+		*str = strcat(*str, buf);
+		*str = strcat(*str, time == 1 ? one : more);
+		if (--(*n) > 1)
+			*str = strcat(*str, ", ");
+		else if (*n == 1)
+			*str = strcat(*str, " and ");
+	}
+}
+
+char	*human_time(unsigned long long seconds)
+{
+	unsigned long long	years;
+	short				days;
+	char				hours, mins;
+	char				*new;
+	char				number_times;
+
+	years = transform_time(&seconds, 31536000);
```

Comment:
This is "subjective" but, `#define`'ing the time units, for example as `#define YEAR 31536000` at the top of the file, would make it easier to read for someone else that it taking a look at your code (obviously here it's obvious you are talking about the amount of seconds in a year, but I'm talking about a more complex program).

Reply:
🐱‍🏍 Normally I do, but that time I didn't 🤷‍♀️🤷‍♀️